### PR TITLE
feat: [DetailsV2] Offer and Listing Tables

### PR DESCRIPTION
**Pull request by cbachmeier**
## Description

- Adds a generic Table content component used by both Listings and Offers table
  - Includes header row, content rows, expandable cell, and specific cell types for common uses such as Prices and Action Buttons
  - Offers uses mock data and Listings uses real data
  - Created a basic Offer object that can be expanded when BE adds offer support
- Moved `Scrim` to shared styles
- Updated max-width of Data page to account for padding
- Pull multiple listings from the Details query
  - tested on V1 page which has no effect as data type was always an array and we just took the first item at index 0 prior
- Adjusted height for data page on below medium screens to add bottom padding
- Added snapshot tests for both Offers and Listings tables
- Added mock offer and listing test util object
- Updated timeSince util to be more generic time util and added timeUntil helper fn with supporting test cases

_JIRA ticket:_ 
 - https://uniswaplabs.atlassian.net/browse/NFT-1172
 - https://uniswaplabs.atlassian.net/browse/NFT-1173
 - https://uniswaplabs.atlassian.net/browse/NFT-1174
_Relevant docs:_ https://www.notion.so/Details-Redesign-0b14fc9f47d946c8af5523b27812205e#6193257f855141d68e7c58552f0e792e

## Screen capture

721 and 1155 Tables for desktop screens
Screen Shot 2023-05-09 at 09 40 55
Screen Shot 2023-05-09 at 09 41 01
Screen Shot 2023-05-09 at 09 41 06
Screen Shot 2023-05-09 at 09 41 13

721 and 1155 Tables for mobile screens
Screen Shot 2023-05-09 at 10 31 58
Screen Shot 2023-05-09 at 10 32 11
Screen Shot 2023-05-09 at 10 32 20
Screen Shot 2023-05-09 at 10 32 26

#### Devices

- Design and I tested on multiple screen widths

### Automated testing

- [x] Unit test for timeUntil util
- [x] Snapshot tests for the two tables

### CODE REVIEW

**Review by JackShort — CHANGES_REQUESTED** [review]

File: src/nft/components/details/detailsV2/DataPageTable.tsx [hunk 1]
```diff
@@ -1,45 +1,50 @@
...
+          TableTabsKeys.Offers,
+          {
+            title: <Trans>Offers</Trans>,
+            key: TableTabsKeys.Offers,
+            content: <OffersTableContent asset={asset} />,
+            count: 11, // TODO Replace Placeholder with real data
+          },
+        ],
+        [
+          TableTabsKeys.Listings,
```
[the remark this answers is not in the document]
Comment (cbachmeier) [reply src/nft/components/details/detailsV2/DataPageTable.tsx:38]: Yes, design confirmed

File: src/nft/components/details/detailsV2/TableContentComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+          </ThemedText.SubHeaderSmall>
+        </TableCell>
+      )}
+      <TableCell $justifyContent="flex-end">
+        <ThemedText.SubHeaderSmall color="textSecondary">
+          <Trans>Expires in</Trans>
+        </ThemedText.SubHeaderSmall>
+      </TableCell>
+      {/* An empty cell is needed in the headers for proper vertical alignment with the action buttons */}
+      <TableCell $flex={isMobile ? 0.25 : 1}>&nbsp;</TableCell>
```
[the remark this answers is not in the document]
Comment (cbachmeier) [reply src/nft/components/details/detailsV2/TableContentComponent.tsx:160]: I tried this originally but DOM wouldn't render the component with nothing in it

File: src/nft/components/details/detailsV2/TableContentComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        </TableCell>
+      )}
+      <TableCell $justifyContent="flex-end">
+        <ThemedText.LabelSmall color="textPrimary">
+          {date ? timeUntil(date) : <Trans>Never</Trans>}
+        </ThemedText.LabelSmall>
+      </TableCell>
+      <TableCell $flex={isMobile ? 0.25 : 1} $justifyContent="center">
+        <ActionButton>
+          {isMobile ? (
```
[the remark this answers is not in the document]
Comment (cbachmeier) [reply src/nft/components/details/detailsV2/TableContentComponent.tsx]: As an alternative I moved this logic up to the parent so the passed in `buttonCTA` will be either the translated text component or the icon.

In general I think a better practice is to use const over let whenever possible to leverage its data immutability.

File: src/nft/utils/time.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import { roundAndPluralize } from './roundAndPluralize'
+
+const SECOND = 1000
+const MINUTE = SECOND * 60
+const HOUR = MINUTE * 60
+const DAY = 24 * HOUR
+const WEEK = 7 * DAY
+const MONTH = 30 * DAY
+
+export function timeUntil(date: Date, originalDate?: Date): string | undefined {
```
[the remark this answers is not in the document]
Comment (cbachmeier) [reply src/nft/utils/time.ts:12]: I like this simplification, however I removed the declaration of `timeLeft`

**Review by JackShort — APPROVED** [review]
lgtm

File: src/nft/components/details/detailsV2/TableRowComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+import { HomeSearchIcon } from 'nft/components/icons'
+import { Offer, SellOrder } from 'nft/types'
+import { formatEth, getMarketplaceIcon, timeUntil } from 'nft/utils'
+import styled from 'styled-components/macro'
+import { BREAKPOINTS, ExternalLink, ThemedText } from 'theme'
+import { shortenAddress } from 'utils'
+
+import { TableTabsKeys } from './DataPageTable'
+
+const TableCell = styled.div<{ $flex?: number; $justifyContent?: string; $color?: string; hideOnSmall?: boolean }>`
```
Comment (cmcewen) [inline src/nft/components/details/detailsV2/TableRowComponent.tsx:20]: i believe it's possible to also just pass a `style={{}}` prop to the component vs explicitly add these css props like justifyContent - is there a reason to prefer either way?

**Review by cmcewen — APPROVED** [review]

File: src/nft/components/details/detailsV2/TableRowComponent.tsx (hunk 5 above)
Comment (cbachmeier) [reply src/nft/components/details/detailsV2/TableRowComponent.tsx:20]: By controlling the props we can enforce both the types of the params and structure of the component wherever it gets implemented. Leaving the component to inline-styling will remove a lot of the rigidity gained from TS styled components

**Review by JackShort — APPROVED** [review]

**Review by JackShort — APPROVED** [review]

**Pull request merged**